src/libraries/Microsoft.PowerFx.Interpreter/Environment/SymbolExtensions.cs

```diff
@@ -31,6 +32,17 @@ public static void SetCulture(this RuntimeConfig symbols, CultureInfo culture)
             symbols.AddService(culture ?? throw new ArgumentNullException(nameof(culture)));
         }
 
+        /// <summary>
+        /// Set UserInfo.
+        /// </summary>
+        /// <param name="symbols">SymbolValues where to set the UserInfo.</param>
+        /// <param name="userInfo">UserInfo to set.</param>
+        /// <exception cref="ArgumentNullException">When userInfo is null.</exception>
+        public static void SetUserInfo(this RuntimeConfig symbols, UserInfo userInfo)
```

Comment:
This should take the interface, IUserInfo.

Reply:
update